### PR TITLE
Allow hiding container contents in inventory

### LINKED ISSUE #45553: Hide/show container's content in inventory view
# Is your feature request related to a problem? Please describe.

Before the nested containers update, we could group certain tools with qualities into one item with more qualities (toolbox).
Now we can manually put certain tools into a container but it's a play-pretend and doing that causes a mess in inventory view.

# Describe the solution you'd like
Keybinding to show/hide container's content in inventory and drop view - things inside TOOLBAG wouldn't be displayed (the way how it works for containers in advanced inventory/pickup view):
image
It would help with medicine-dedicated containers too, I always carry certain meds and bandages but inventory view shuffles them with random `DRUGS` I looted.

# Describe alternatives you've considered
Option to bind items in a container and the container into a single piece:

- Its content wouldn't be displayed
- It could be `a`ctivated to enter activation menu where we would only see its content. Basically the way tool belt, survivor's belt and toolbox used to work. 
- No item can be inserted or removed until unbinding.

image

# Additional content
image

### PR DESCRIPTION
**Pull request by RoyBerube**
#### Summary
Interface "Allow hiding container contents in inventory"

#### Purpose of change

Fixes #45553

Quality of life UI improvement to allow the player to hide the contents of containers in inventory. MRE and first aid kits can cause a lot of clutter; this gives the player the option to hide the contents.

#### Describe the solution

The player can toggle a container to show or hide its contents using the < and > keys. The container will change its display to indicate the status - see screenshots below. The player setting will persist between saves or when dropped on the ground and picked up again.

Also added the ability to activate or consume just from that container.  Done from the item action menu.

#### Testing

Full test program: no errors.

Tested with assorted kits, backpacks, tools and weapons that use ammo, and clothing with pockets.
Behavior is working as designed.

#### What this PR does not do:

No items currently have a default collapsed setting.

Toggling the top container of nested containers will hide or show everything. Some way to keep user hidden containers from automatically showing this way could be a feature to add eventually.

#### Additional context

Default display with no containers hidden.
<br/>
<br/>
<br/>

Messenger bag hidden.
<br/>
<br/>
<br/>

All items in inventory are hidden.
<br/>
<br/>
<br/>

Messenger bag action menu.
<br/>
<br/>
<br/>

Eat the messenger bag in action menu shows it consumable contents.
<br/>
<br/>
<br/>

Activate the jeans in action menu.

### CODE REVIEW

**Comment by RDru** [comment]
Let me suggest (probably as another PR) to hide by default all items in containers in **w**ield, **W**ear and (maybe) **U**nload menus.

**Comment by RoyBerube** [comment]
Another PR most likely.

I'm also thinking of adding a flag to some items to make them collapsed by default. MREs, first aid kits, and wallets come to mind. That can come later too.

**Comment by actual-nh** [comment]
@RoyBerube: Pinging you regarding the conflicts; dunno if GitHub notifies you for those.

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/easy-way-to-empty-wallet/26598/13

**Comment by l29ah** [comment]
This change results in https://github.com/CleverRaven/Cataclysm-DDA/issues/51807 and probably other issues.

File: src/game.cpp [hunk 1]
```diff
@@ -1472,6 +1472,23 @@ static hint_rating rate_action_disassemble( avatar &you, const item &it )
...
 {
+    if( it.is_container() ) {
+        hint_rating best_rate = hint_rating::cant;
+        it.visit_items( [&you, &best_rate]( item * node, item * ) {
+            if( you.can_consume( *node ) )  {
+                ret_val<edible_rating> rate = you.will_eat( *node );
+                if( rate.success() ) {
+                    best_rate = hint_rating::good;
+                    return VisitResponse::ABORT;
+                } else if( rate.value() != INEDIBLE || rate.value() != INEDIBLE_MUTATION ) {
```
Comment (ZhilkinSerg) [inline src/game.cpp:1483]: This expression (NOT `X` OR NOT `B`) would always be true. Did you want to used `&&` operator instead of `||` in here?

**Pull request merged**